scipy/signal/_signaltools.py

```diff
@@ -2398,16 +2487,22 @@ def hilbert(x, N=None, axis=-1):
     >>> fig.tight_layout()
 
     """
-    x = np.asarray(x)
-    if np.iscomplexobj(x):
+    xp = array_namespace(x)
+
+    x = xp.asarray(x)
+    if xp.isdtype(x.dtype, 'complex floating'):
         raise ValueError("x must be real.")
+
+    # XXX https://github.com/scipy/scipy/issues/21507
+    x = xp.astype(x, xp.complex64 if x.dtype == xp.float32 else xp.complex128)
```

Comment:
this issue is closed now

Reply:
Yes, thanks. With an additional fix on top of https://github.com/scipy/scipy/pull/21508 to account for integer inputs, `test_hilbert_axisN[array_api_strict]` finally passes.